File: 1768_merge_strings_alternately/Solution.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        std::string merged = "";
        
        for(int i = 0; i<word1.size(); i++){
            if(i == word2.size()){
                merged += word1.substr(i,word1.size());
                return merged;
            }
            if(i == word1.size() - 1 && word2.size() > word1.size()){
                merged += word1.substr(i,1) + word2.substr(i, word2.size());
                return merged;
            }
            merged += word1.substr(i,1) + word2.substr(i,1);
        }
        
        return merged;
    }
};

File: 1768_merge_strings_alternately/Solution.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        String merged = "";
        
        for(int i = 0; i<word1.length(); i++){
            if(i == word2.length()){
                return merged + word1.substring(i);
            }
            if(i == word1.length() - 1 && word2.length() > word1.length()){
                return merged + word1.substring(i) + word2.substring(i);
            }
            merged += word1.substring(i,i+1) + word2.substring(i,i+1);
        }
        return merged;
    }
}
